tests/components/apprise/test_notify.py

```diff
@@ -0,0 +1,68 @@
+"""The tests for the apprise notification platform."""
+from unittest.mock import patch
+
+from homeassistant.setup import async_setup_component
+
+BASE_COMPONENT = "notify"
+
+
+async def test_apprise_config_load_fail(hass):
+    """Test apprise configuration failures."""
+
+    config = {"apprise": {"name": "test", "platform": "apprise", "config": "/path/"}}
+
+    with patch("apprise.AppriseConfig.add", return_value=False):
+        assert await async_setup_component(hass, BASE_COMPONENT, config)
+        await hass.async_block_till_done()
+
+    with patch("apprise.AppriseConfig.add", return_value=True):
+        with patch("apprise.Apprise.add", return_value=False):
+            assert await async_setup_component(hass, BASE_COMPONENT, config)
+            await hass.async_block_till_done()
+
+
+async def test_apprise_config_load_okay(hass, tmpdir):
+    """Test apprise configuration failures."""
+
+    # Test cases where our URL is invalid
+    t = tmpdir.mkdir("apprise-config").join("apprise")
+    t.write("mailto://user:pass@example.com/")
+
+    config = {"apprise": {"name": "test", "platform": "apprise", "config": str(t)}}
+
+    assert await async_setup_component(hass, BASE_COMPONENT, config)
+    await hass.async_block_till_done()
+
+
+async def test_apprise_url_load_fail(hass):
+    """Test apprise url failure."""
+
+    config = {
+        "apprise": {
+            "name": "test",
+            "platform": "apprise",
+            "url": "mailto://user:pass@example.com",
+        }
+    }
+    with patch("apprise.Apprise.add", return_value=False):
+        assert await async_setup_component(hass, BASE_COMPONENT, config)
+        await hass.async_block_till_done()
+
+
+async def test_apprise_notification(hass):
+    """Test apprise notification."""
+
+    config = {
+        "apprise": {
+            "name": "test",
+            "platform": "apprise",
+            "url": "mailto://user:pass@example.com",
+        }
+    }
+    data = {"title": "Test Title", "message": "Test Message"}
+    with patch("apprise.Apprise.notify", return_value=True):
```

Comment:
```suggestion
    with patch("homeassistant.components.apprise.notify.Apprise.notify", return_value=True):
```

Reply:
The adaptation for Apprise into Home Assistant doesn't really require a lot of testing since 99.999% of the testing is already covered in Apprise (which has 100% testing coverage). My initial goal for test cases were to just pass through all of the sections of new code I added.  there is no magic taking place otherwise to test.

On a side note (for this specific test): Strangely the **patch()** works fantastic as is on everything but the **Apprise.notify()** function. It executes the command in both the old way and your new suggested way.  It seemed to work when i just mocked the entire **Apprise()** which is what the latest merge has. Let me know if you still want me to explicitly access the imported module instead of the general library (both achieve the same results).